testing/file_test/file_test_base.cpp

```diff
@@ -213,10 +213,42 @@ auto main(int argc, char** argv) -> int {
     return EXIT_FAILURE;
   }
 
-  std::string subset_target_storage = GetSubsetTarget();
-  ::subset_target = &subset_target_storage;
+  const char* target = getenv("TEST_TARGET");
+  CARBON_CHECK(target != nullptr);
 
-  std::vector<llvm::StringRef> paths(argv + 1, argv + argc);
+  // Configure the name of the subset target.
+  std::string subset_target_storage = target;
+  static constexpr char SubsetSuffix[] = ".subset";
+  if (!llvm::StringRef(subset_target_storage).ends_with(SubsetSuffix)) {
+    subset_target_storage += SubsetSuffix;
+  }
+  Carbon::Testing::subset_target = &subset_target_storage;
+
+  // Save the working directory for later restoration.
+  std::error_code ec;
+  std::filesystem::path orig_working_dir_storage =
+      std::filesystem::current_path(ec);
+  CARBON_CHECK(!ec) << ec.message();
+  Carbon::Testing::orig_working_dir = &orig_working_dir_storage;
+
+  // Configure the base directory for test names.
+  llvm::StringRef target_dir = target;
+  // Leaves one slash.
+  CARBON_CHECK(target_dir.consume_front("/"));
+  target_dir = target_dir.substr(0, target_dir.rfind(":"));
+  std::string base_dir =
+      orig_working_dir_storage.string() + target_dir.str() + "/";
+  Carbon::Testing::base_dir_len = base_dir.size();
```

Comment:
Given this code (which I'm totally fine with), not sure using `std::filesystem::path` everywhere is really adding much as everything is really manipulated as strings. I might just immediately switch to strings after the `current_path` and `absolute` calls, and keep all the APIs in terms of `StringRef` where possible, but your call. I'm fine either way.

Reply:
The reason I was propagating std::filesystem::path is because I also use both `.filename()` and `.parent_path()`. If I kept APIs as `StringRef`, I'd either need to implement both myself or essentially do something like `std::filesystem::path(StringRef.str()).filename()`, creating more string copies rather than reducing them. This approach just felt like the right balance of trade-offs, given the `absolute` call produces a `path`.